btree/btree.c

```diff
@@ -2839,18 +2912,35 @@ static void generic_move(struct nd *src, struct nd *tgt,
 	node_fix(src, tx);
 	node_seq_cnt_update(tgt);
 	node_fix(tgt, tx);
+}
 
-	/**
-	 * ToDo: We need to capture the changes occuring in the "src" node
-	 * using m0_be_tx_capture().
-	 * Only the modified memory from the node needs to be updated.
-	 */
+static bool ff_capture(struct slot *slot, struct m0_be_tx *tx)
+{
+	struct ff_head   *h     = ff_data(slot->s_node);
+	int               rsize  = h->ff_ksize + h->ff_vsize;
+	void             *start = ff_key(slot->s_node, slot->s_idx);
+	struct m0_be_seg *seg   = slot->s_node->n_tree->t_seg;
+	m0_bcount_t       hsize = sizeof(*h) - sizeof(h->ff_opaque);
 
+	rsize = rsize;
 	/**
-	 * ToDo: We need to capture the changes occuring in the "tgt" node
-	 * using m0_be_tx_capture().
-	 * Only the modified memory from the node needs to be updated.
+	 *  Capture starting from the location where new record may have been
+	 *  added or deleted. Capture till the last record. If the deleted
+	 *  record was at the end then no records need to be captured only the
+	 *  header modifications need to be persisted.
 	 */
+	if (h->ff_used > slot->s_idx)
+		M0_BTREE_TX_CAPTURE(tx, seg, start, rsize * (h->ff_used - slot->s_idx));
+	else if (h->ff_opaque == NULL)
+		/**
+		 *  This will happen when the node is initialized in which case
+		 *  we want to capture the opaque pointer.
+		 */
+		hsize += sizeof(h->ff_opaque);
+
+	M0_BTREE_TX_CAPTURE(tx, seg, h, hsize);
+
+	return true;
```

Comment:
We are always returning true. If there is no error condition, change return type to void.

Reply:
Yes I will make this change.